crates/nu-std/std-rfc/iter/mod.nu

```diff
@@ -172,3 +172,50 @@ export def recurse [
     generate $fn [{path: ($.), item: ($in) }]
     | if not $depth_first { flatten } else { }
 }
+
+# Helper for `only` errors
+def only-error [msg: string, meta: record, label: string]: nothing -> error {
+  error make {
+    msg: $msg,
+    label: {
+      text: $label,
+      span: $meta.span,
+    }
+  }
+}
+
+# Get the only element of a list or table, ensuring it exists and there are no extra elements.
+#
+# Similar to `first` with no arguments, but errors if there are no additional
+# items when there should only be one item. This can help avoid issues when more
+# than one row than expected matches some criteria.
+#
+# This command is useful when chained with `where` to ensure that only one row
+# meets the given condition.
+#
+# If a cell path is provided as an argument, it will be accessed after the first
+# element. For example, `only foo` is roughly equivalent to `get 0.foo`, with
+# the guarantee that there are no additional elements.
+#
+# Note that this command currently collects streams.
+@search-terms "first" "single"
+@example "Get the only item in a list, ensuring it exists and there's no additional items" --result 5 {
```

Comment:
nice to have a `@category` too on this (probably need it on many stdlib custom commands.